server/service/src/sync/sync_on_central/mod.rs

```diff
@@ -0,0 +1,116 @@
+    // AND
+    // (
+    // 	table_name in {central_record_names}
+    //  OR
+    // 	(table_name in {transfer record names}  AND name_id IN {name_ids of active stores on remote site})
+    //  OR
+    // 	// Special cases
+    // 	(table_name in {patient record name} AND patient_id IN {select name_id from name_store_join where store_id in {active stores on remote site})
+    // )
+    // When we upgrade to diesel 2 we can do dynaimc filter: https://github.com/andreievg/diesel-rs-dynamic-filters
+    // And the above would become something like:
+    // use ChangeLog::Filter as f;
+    // let filter = create_and_filter(vec![
+    //     f::cursor(NumberFilter::GreaterThen(cursor)),
+    //     f::last_sync_site_id(NumberFilter::NotEqual(site_id)),
+    //     f::Or(
+    //         vec![f::table_name(StringFilter::Any(central_record_table_names))],
+    //         f::And(vec![
+    //             f::table_name(StringFilter::Any(transfer_record_names)),
+    //             f::name_id(StringFilter::Any(active_stores_on_site.name_ids())),
+    //         ]),
+    //         f::And(vec![
+    //             f::table_name(StringFilter::Any(remote_record_names)),
+    //             f::store_id(StringFilter::Any(active_stores_on_site.store_ids())),
+    //         ]),
+    //         f::And(vec![
+    //             f::table_name(StringFilter::Any(patient_record_names)),
+    //             f::patient_id(StringFilter::Any(
+    //                 name_store_joins::create_filtered_query(name_store_join::Filter::store_id(
+    //                     StringFitler::Any(active_stores_on_site.store_ids()),
+    //                 ))
+    //                 .select(name_store_joins::name_id),
+    //             )),
+    //         ]),
+    //         // Etc..
+    //     ),
+    // ]);
+}
+
+pub async fn pull(
+    service_provider: &ServiceProvider,
+    SyncRequestV6 {
+        cursor,
+        batch_size,
+        sync_v5_settings,
+    }: SyncRequestV6,
+) -> Result<SyncBatchV6, SyncParsedErrorV6> {
+    use SyncParsedErrorV6 as Error;
+
+    if !is_central_server() {
+        return Err(Error::NotACentralServer);
+    }
+    // Check credentials again mSupply central server
+    SyncApiV5::new(sync_v5_settings)
+        .map_err(|e| Error::OtherServerError(format_error(&e)))?
+        .get_site_status()
+        .await
+        .map_err(Error::from)?;
+
+    // TODO Versioning ?
+
+    let ctx = service_provider.basic_context()?;
+    // To push or not ?
+    let changelog_repo = ChangelogRepository::new(&ctx.connection);
+
+    let filter = Some(create_filter());
+
+    let changelogs = changelog_repo.changelogs(cursor, batch_size, filter.clone())?;
+    let total_records = changelog_repo.count(cursor, filter)?;
+    let max_cursor = changelog_repo.latest_cursor()?;
+
+    let end_cursor = changelogs
+        .last()
+        .map(|log| log.cursor as u64)
+        .unwrap_or(max_cursor);
+
+    let records = translate_changelogs_to_push_records(
```

Comment:
It feels a bit weird that this function is called `Pull` but creates `Push` records.
Maybe the function could be called `pull_from_remote` that does sound a bit clunky though.
This is just me sharing my thoughts/feedback, not suggesting we need to change it...

Reply:
I was hoping the OmSupplyCentralSitePush type would make sense (plus the comments for that variant). 

I also found it confusing.

How about I rename translators and associated methods to: "from_sync_record", "to_sync_record" ?

Reply:
And of course add some comment to the "pull" method here to say it's used by central API in response to a pull records request from remote site